Core/Persistence/Legacy/Content/FieldValue/Converter/EnhancedSelectionConverter.php

```diff
@@ -63,6 +119,39 @@ public function toStorageFieldDefinition( FieldDefinition $fieldDef, StorageFiel
      */
     public function toFieldDefinition( StorageFieldDefinition $storageDef, FieldDefinition $fieldDef )
     {
+        $simpleXml = simplexml_load_string( $storageDef->dataText5 );
+        $options = array();
+        $isMultiple = false;
+        $delimiter = false;
+        $query = false;
+
+        if ( $simpleXml !== false )
+        {
+            foreach ( $simpleXml->options->option as $option )
+            {
+                $options[(int)$option["id"]] = array( "id" => (string)$option["id"],
+                                                      "name" => (string)$option["name"],
+                                                      "identifier" => (string)$option["identifier"],
+                                                      "priority" => (string)$option["priority"] );
+            }
+            if( $simpleXml->multiselect == 1 )
+            {
+                $isMultiple = true;
+            }
+            if( $simpleXml->query )
+            {
+                $query = (string)$simpleXml->query;
+            }
+        }
+
+        $fieldDef->fieldTypeConstraints->fieldSettings = new FieldSettings(
+            array(
+                "isMultiple" => $isMultiple,
+                "delimiter" => $delimiter,
```

Comment:
Hello @emodric 

We have addressed this issue with a new commit. Thank you for pointing out this bug. For some reason GitHub is not changing this code comment block as outdated so we made this comment.